portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -85,12 +105,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       userName = "Anonymous";
     }
     long timestamp = System.currentTimeMillis();
+    String userId = userService.getCurrentUser().getUserId();
+    String email = userService.getCurrentUser().getEmail();
+    AuthenticationServlet.updateUserName(userName, userId);
 
     if (!Strings.isNullOrEmpty(userComment)) {    
       DatastoreService datastore = 
         DatastoreServiceFactory.getDatastoreService();
-      datastore.put((new Comment(userComment, userName, timestamp, 0))
+      Key key = datastore.put(
+        (new Comment(userComment, "", timestamp, 0, userId, email))
         .toEntity());
+      try {
```

Comment:
Why need this? Isn't the comment already added with ID when you first call datastore.put?

Reply:
You're totally right; this whole structure is definitely not ideal! I spent a long time scouring the documentation to try to find a better way, and basically what I discovered is that, even though entities are stored with a unique id, not only can you not use the id of an Entity with datastore.get() (or somehow use the id to get the Key it corresponds to as far as I was able to find), it looks like you can't filter by that unique id when querying the database, like I do in DeleteServlet:

`Query query = new Query("Comment").setFilter(new Query.FilterPredicate
        (EntityProperties.COMMENT_ID, Query.FilterOperator.EQUAL, id));`
  `Entity commentWithCorrectId = datastore.prepare(query).asSingleEntity();`

As far as I was able to glean from the documentation, you can only filter by either properties you've created or the key if you have it. It seems super redundant to have a "ID/Name" and "id" field that store the exact same value, but I made the "id" property because I needed an id field I could access while filtering my query. 

All that being said, I'm very open to suggestions on how to handle this functionality better :)

Reply:
let's discuss in our 1:1 :)